test/protocol-tests-core/src/main/java/software/amazon/awssdk/protocol/ProtocolTestSuiteLoader.java

```diff
@@ -51,11 +51,13 @@ private TestSuite loadTestSuite(String suitePath) throws IOException {
     private Stream<? extends TestCase> loadTestCases(String testCase) {
         try {
             final List<TestCase> testCases = MAPPER
-                    .readValue(getClass().getResource(RESOURCE_PREFIX + testCase), new TypeReference<List<TestCase>>() {
-                    });
+                    .readValue(getClass().getResource(RESOURCE_PREFIX + testCase), new ListTypeReference());
             return testCases.stream();
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
+
+    private static class ListTypeReference extends TypeReference<List<TestCase>> {
```

Comment:
This seems excessive.

Reply:
Maybe a bit, yeah, but I can see its point that maybe we shouldn't be leaking a reference to the parent if we don't need to. Eg. If Jackson does any kind of caching of its unmarshallers that causes it to hold onto the type reference. I'm fairly sure it doesn't in this case, since it's mostly a mechanism for delivering type information.

Reply:
Wish the compiler could do this crap for us. I.E. detect it doesn't need a reference to this and convert it to a static class.

Reply:
Yeah, or provide a way to make an an anonymous static inner class.